cli/internal/cmd/trackingplan/apply/apply.go

```diff
@@ -39,21 +42,25 @@ func NewCmdTPApply() *cobra.Command {
 			$ rudder-cli tp apply --loc </path/to/dir or file> --dry-run
 		`),
 		PreRunE: func(cmd *cobra.Command, args []string) error {
+			fmt.Println()
+
 			// Here we might need to do validate
 			localcatalog, err = readCatalog(catalogDir)
 			if err != nil {
 				return fmt.Errorf("reading catalog failed in pre-step: %w", err)
 			}
 
-			if err := validate.ValidateCatalog(validate.DefaultValidators(), localcatalog); err != nil {
-				return fmt.Errorf("validating catalog: %w", err)
+			if errs := validate.NewCatalogValidator().Validate(localcatalog); errs != nil {
+				fmt.Println("catalog errors:")
+				fmt.Println(errs.Error())
+
```

Comment:
It looks like we're reporting errors in both apply and validate commands. It's better if reporting the validation results is moved to a dedicated function (in Validator itself maybe?) and use that in both places.